### DISCUSSION TITLE
Sorting Results

### DISCUSSION THREAD

**Discussion by Jonatha1983** [community/community#10587] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi
I think the ability of sorting the results is really missing. 
A sort by Stars / Updated (indexed)  is really helpful and maybe Languages / ALPHABETIC / FILE Types 

Thanks

**Comment by colinwm** [comment, marked as answer]
Thanks for the feedback @Jonatha1983. We currently sort results by our scoring algorithm, which takes into account stars, query relevance and some other factors. And we also allow you to filter down to specific languages (try adding `language:python` to your query to limit to python code only, for example)

Can you explain why you'd want to get results in alphabetical order? Or ordered by last indexed? What would you use that for?

**Comment by Jonatha1983** [comment]
Hi

Thanks for the super quick response

About the sort by last update / indexed -  very helpful for cases I am
using some advanced code usage and I would like to know that what I am
searching for was updated recently.
Or also the opposite in case of some old code usage.
For example if I am searching examples of new gradle feature I would like to sort by updated so I am sure I will see relevant results. 

About the Alphabetical Order helps when the results contains more than one
results I am interesting in and so it help looking at them in a certain way

Also I could not found a way to do some more complex search like: find
files named: build.gradle with both text: A and B - maybe I miss it in the
documentation but will glad to get a reference.

Thanks again

**Reply by colinwm** [reply]
> Also I could not found a way to do some more complex search like: find files named: build.gradle with both text: A and B - maybe I miss it in the documentation but will glad to get a reference.

You can read about the syntax here: https://cs.github.com/about/syntax

To do a search for e.g. a particular file and two terms, just do something like:

```
path:build.gradle A B 
```

**Reply by Jonatha1983** [reply]
Yes thanks found it after I posted the reply (sorry)
